(cross-)compiling with armv7-unknown-linux-gnueabihf fails

(cross-)compiling with target `armv7-unknown-linux-gnueabihf` fails with `type mismatch` errors.

## Expected Behavior

(cross-)compiling with target `armv7-unknown-linux-gnueabihf` succeeds.

## Actual Behavior

(cross-)compiling with target `armv7-unknown-linux-gnueabihf` fails with the following errors

```
error[E0631]: type mismatch in function arguments
   --> src/backend/winit.rs:163:74
    |
163 |         } else if let Some(xlib_window) = winit_window.xlib_window().map(native::XlibWindow) {
    |                                                                          ^^^^^^^^^^^^^^^^^^ expected signature of `fn(u32) -> _`
    | 
   ::: src/backend/egl/native.rs:229:1
    |
229 | pub struct XlibWindow(pub u64);
    | ------------------------------- found signature of `fn(u64) -> _`

error[E0308]: mismatched types
   --> src/wayland/dmabuf/mod.rs:442:38
    |
442 |             if plane.offset as i64 > size {
    |                                      ^^^^
    |                                      |
    |                                      expected `i64`, found `i32`
    |                                      help: you can convert an `i32` to an `i64`: `size.into()`

error[E0308]: mismatched types
   --> src/wayland/dmabuf/mod.rs:449:55
    |
449 |             if (plane.offset + plane.stride) as i64 > size {
    |                                                       ^^^^
    |                                                       |
    |                                                       expected `i64`, found `i32`
    |                                                       help: you can convert an `i32` to an `i64`: `size.into()`

error[E0308]: mismatched types
   --> src/wayland/dmabuf/mod.rs:458:53
    |
458 |             if plane.plane_idx == 0 && end as i64 > size {
    |                                                     ^^^^
    |                                                     |
    |                                                     expected `i64`, found `i32`
    |                                                     help: you can convert an `i32` to an `i64`: `size.into()`

error: aborting due to 4 previous errors
```

## Steps to Reproduce the Problem

(cross-)compile anvil with target `armv7-unknown-linux-gnueabihf`

## Specifications

  - Version: master
  - Platform: `armv7-unknown-linux-gnueabihf`

--- a/src/backend/egl/native.rs
+++ b/src/backend/egl/native.rs
@@ -2,6 +2,7 @@
 
 use super::{display::EGLDisplayHandle, ffi, wrap_egl_call, SwapBuffersError};
 use nix::libc::{c_int, c_void};
+use std::os::raw;
 #[cfg(feature = "backend_gbm")]
 use std::os::unix::io::AsRawFd;
 use std::{fmt::Debug, marker::PhantomData, sync::Arc};
@@ -226,7 +227,7 @@ pub unsafe trait EGLNativeSurface: Send + Sync {
 
 #[cfg(feature = "backend_winit")]
 /// Typed Xlib window for the `X11` backend
-pub struct XlibWindow(pub u64);
+pub struct XlibWindow(pub raw::c_ulong);
 
 #[cfg(feature = "backend_winit")]
 unsafe impl EGLNativeSurface for XlibWindow {
@@ -241,7 +242,7 @@ unsafe impl EGLNativeSurface for XlibWindow {
             ffi::egl::CreatePlatformWindowSurfaceEXT(
                 display.handle,
                 config_id,
-                (&mut id) as *mut u64 as *mut _,
+                (&mut id) as *mut raw::c_ulong as *mut _,
                 surface_attributes.as_ptr(),
             )
         })

--- a/src/wayland/dmabuf/mod.rs
+++ b/src/wayland/dmabuf/mod.rs
@@ -439,14 +439,14 @@ fn buffer_basic_checks(
         if let Ok(size) = ::nix::unistd::lseek(plane.fd.unwrap(), 0, ::nix::unistd::Whence::SeekEnd) {
             // reset the seek point
             let _ = ::nix::unistd::lseek(plane.fd.unwrap(), 0, ::nix::unistd::Whence::SeekSet);
-            if plane.offset as i64 > size {
+            if plane.offset as libc::off_t > size {
                 params.as_ref().post_error(
                     ParamError::OutOfBounds as u32,
                     format!("Invalid offset {} for plane {}.", plane.offset, plane.plane_idx),
                 );
                 return false;
             }
-            if (plane.offset + plane.stride) as i64 > size {
+            if (plane.offset + plane.stride) as libc::off_t > size {
                 params.as_ref().post_error(
                     ParamError::OutOfBounds as u32,
                     format!("Invalid stride {} for plane {}.", plane.stride, plane.plane_idx),
@@ -455,7 +455,7 @@ fn buffer_basic_checks(
             }
             // Planes > 0 can be subsampled, in which case 'size' will be smaller
             // than expected.
-            if plane.plane_idx == 0 && end as i64 > size {
+            if plane.plane_idx == 0 && end as libc::off_t > size {
                 params.as_ref().post_error(
                     ParamError::OutOfBounds as u32,
                     format!(

